docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,409 @@
+
+We generate the description in this approach based on `spec.customresourcedefinitions` present in the ClusterServiceVersion.
+This CRD def is different from what is provided by the `CustomResourceDefinition` as it has less information.
+
+This is how one of the `customresourcedefinition` looks like ( Kafka from strimzi )
+```
+{
+  "parameters": [
+    {
+      "description": "Kafka version",
+      "displayName": "Version",
+      "path": "kafka.version",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:text"
+      ]
+    },
+    {
+      "description": "The desired number of Kafka brokers.",
+      "displayName": "Kafka Brokers",
+      "path": "kafka.replicas",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:podCount"
+      ]
+    },
+    {
+      "description": "The type of storage used by Kafka brokers",
+      "displayName": "Kafka storage",
+      "path": "kafka.storage.type",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:select:ephemeral",
+        "urn:alm:descriptor:com.tectonic.ui:select:persistent-claim",
+        "urn:alm:descriptor:com.tectonic.ui:select:jbod"
+      ]
+    },
+    {
+      "description": "Limits describes the minimum/maximum amount of compute resources required/allowed",
+      "displayName": "Kafka Resource Requirements",
+      "path": "kafka.resources",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:resourceRequirements"
+      ]
+    },
+    {
+      "description": "The desired number of Zookeeper nodes.",
+      "displayName": "Zookeeper Nodes",
+      "path": "zookeeper.replicas",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:podCount"
+      ]
+    },
+    {
+      "description": "The type of storage used by Zookeeper nodes",
+      "displayName": "Zookeeper storage",
+      "path": "zookeeper.storage.type",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:select:ephemeral",
+        "urn:alm:descriptor:com.tectonic.ui:select:persistent-claim",
+        "urn:alm:descriptor:com.tectonic.ui:select:jbod"
+      ]
+    },
+    {
+      "description": "Limits describes the minimum/maximum amount of compute resources required/allowed",
+      "displayName": "Zookeeper Resource Requirements",
+      "path": "zookeeper.resources",
+      "x-descriptors": [
+        "urn:alm:descriptor:com.tectonic.ui:resourceRequirements"
+      ]
+    }
+  ]
+}
+```
+
+## Allowing user to set the parameters 
+
+The current approach of sending map values via cli are as follows - 
+
+- we would add a `-p` cobra param as a list
+- each of the parameter would represent the key in a map and value in a map `e.g. -p "key"="value"`
+- we would allow json path in the key for the user to specific any field in the map that they want to set e.g. `services[0].namespace`.
+- we will pass all values as string. 
```

Comment:
I thought that we already verified that this can't be done.
We need to at least detect and recognize the basic types like int, float, bool,  string

Reply:
yeah this part is older

Reply:
fixed